src/test/rpc_wallet_tests.cpp

```diff
@@ -439,6 +478,544 @@ BOOST_AUTO_TEST_CASE(rpc_wallet_z_importexport)
+    BOOST_CHECK_EQUAL(gCounter.load(), numSuccess);
+    BOOST_CHECK(q->getOperationCount() == 0);
+    ids = q->getAllOperationIds();
+    BOOST_CHECK(ids.size()==0);
+}
+
+// This tests z_getoperationstatus, z_getoperationresult, z_listoperationids
+BOOST_AUTO_TEST_CASE(rpc_z_getoperations)
+{
+    std::shared_ptr<AsyncRPCQueue> q = getAsyncRPCQueue();
+    std::shared_ptr<AsyncRPCQueue> sharedInstance = AsyncRPCQueue::sharedInstance();
+    BOOST_CHECK(q == sharedInstance);
+
+    BOOST_CHECK_NO_THROW(CallRPC("z_getoperationstatus"));
+    BOOST_CHECK_NO_THROW(CallRPC("z_getoperationstatus []"));
+    BOOST_CHECK_NO_THROW(CallRPC("z_getoperationstatus [\"opid-1234\"]"));
+    BOOST_CHECK_THROW(CallRPC("z_getoperationstatus [] toomanyargs"), runtime_error);
+    BOOST_CHECK_THROW(CallRPC("z_getoperationstatus not_an_array"), runtime_error);
+
+    BOOST_CHECK_NO_THROW(CallRPC("z_getoperationresult"));
+    BOOST_CHECK_NO_THROW(CallRPC("z_getoperationresult []"));
+    BOOST_CHECK_NO_THROW(CallRPC("z_getoperationresult [\"opid-1234\"]"));
+    BOOST_CHECK_THROW(CallRPC("z_getoperationresult [] toomanyargs"), runtime_error);
+    BOOST_CHECK_THROW(CallRPC("z_getoperationresult not_an_array"), runtime_error);
+    
+    std::shared_ptr<AsyncRPCOperation> op1 = std::make_shared<AsyncRPCOperation>();
+    q->addOperation(op1);
+    std::shared_ptr<AsyncRPCOperation> op2 = std::make_shared<AsyncRPCOperation>();
+    q->addOperation(op2);
+    
+    BOOST_CHECK(q->getOperationCount() == 2);
+    BOOST_CHECK(q->getNumberOfWorkers() == 0);
+    q->addWorker();
+    BOOST_CHECK(q->getNumberOfWorkers() == 1);
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+    BOOST_CHECK(q->getOperationCount() == 0);
+    
+    Value retValue;
+    BOOST_CHECK_NO_THROW(retValue = CallRPC("z_listoperationids"));
+    BOOST_CHECK(retValue.get_array().size() == 2);
+
+    BOOST_CHECK_NO_THROW(retValue = CallRPC("z_getoperationstatus"));
+    Array array = retValue.get_array();
+    BOOST_CHECK(array.size() == 2);
+
+    // idempotent
+    BOOST_CHECK_NO_THROW(retValue = CallRPC("z_getoperationstatus"));
+    array = retValue.get_array();
+    BOOST_CHECK(array.size() == 2);   
+    
+    for (Value v : array) {
+        Object obj = v.get_obj();
+        Value id = find_value(obj, "id");
+        
+        Value result;
+        // removes result from internal storage
+        BOOST_CHECK_NO_THROW(result = CallRPC("z_getoperationresult [\"" + id.get_str() + "\"]"));
+        Array resultArray = result.get_array();
+        BOOST_CHECK(resultArray.size() == 1);
+        
+        Object resultObj = resultArray.front().get_obj();
+        Value resultId = find_value(resultObj, "id");
+        BOOST_CHECK_EQUAL(id.get_str(), resultId.get_str());
+    }
+    
+    // operations removed
+    BOOST_CHECK_NO_THROW(retValue = CallRPC("z_getoperationstatus"));
+    array = retValue.get_array();
+    BOOST_CHECK(array.size() == 0);
+
+    q->close();
```

Comment:
Adding a another test to call z_getoperationresult again and verify the operation has been removed.